lib/utils/validateObjectWithProps.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const { isPlainObject } = require('is-plain-object');
```

Comment:
[suggest] Please use `./validateTypes` instead of `is-plain-object`. The former improves the type inference.

```suggestion
const { isPlainObject } = require('./validateTypes');
```